presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -1111,6 +1113,13 @@ core::JoinType toJoinType(protocol::JoinType type) {
 
   VELOX_UNSUPPORTED("Unknown join type");
 }
+
+inline bool isNestedLoopJoin(const std::shared_ptr<protocol::JoinNode> node) {
+  auto joinType = toJoinType(node->type);
+  return node->criteria.empty() || core::isInnerJoin(joinType) ||
```

Comment:
@karteekmurthys The condition is just node->criteria.empty(). The rest of the clauses catch all joinTypes. Unless I am missing something.